In [1]:
print("Jupyter is working")

Jupyter is working


In [2]:
import pandas as pd

df = pd.read_csv("data/RYA22-24 subset250 data preproc v1.csv")

df.head()

,age,grade,cantril,ses,genderN,cantrilCat,year,gradeCatps,gradeCat,num
0,13,8,1,High,Male,Struggling,2022.0,Grades 7-8,secondary school,1
1,11,6,5,Mid,Male,Doing OK,2022.0,Grades 5-6,primary school,2
2,11,6,8,Mid,Female,Thriving,2022.0,Grades 5-6,primary school,3
3,13,8,5,NaN,Female,Doing OK,2022.0,Grades 7-8,secondary school,4
4,10,4,8,NaN,Male,Thriving,2022.0,Grades 3-4,primary school,5


In [3]:
print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

print("\nColumn names:")
print(df.columns.tolist())

print("\nMissing values:")
print(df.isnull().sum())

Rows: 1000
Columns: 10

Column names:
['age', 'grade', 'cantril', 'ses', 'genderN', 'cantrilCat', 'year', 'gradeCatps', 'gradeCat', 'num']

Missing values:
age             0
grade           0
cantril         0
ses           327
genderN         0
cantrilCat      0
year          250
gradeCatps      0
gradeCat        0
num             0
dtype: int64


In [4]:
print("Year:")
print(df["year"].value_counts(dropna=False))

print("\nSES:")
print(df["ses"].value_counts(dropna=False))

print("\nMissing year and SES together:")
print(df[df["year"].isna()]["ses"].value_counts(dropna=False))

Year:
year
2022.0    250
2023.0    250
2024.0    250
NaN       250
Name: count, dtype: int64

SES:
ses
NaN     327
High    286
Mid     250
Low     137
Name: count, dtype: int64

Missing year and SES together:
ses
NaN    250
Name: count, dtype: int64


In [5]:
print("Missing SES by year:")
print(df[df["ses"].isna()]["year"].value_counts(dropna=False))

print("\nRows with missing year:")
display(df[df["year"].isna()].head(10))

Missing SES by year:
year
NaN       250
2022.0     26
2024.0     26
2023.0     25
Name: count, dtype: int64

Rows with missing year:


,age,grade,cantril,ses,genderN,cantrilCat,year,gradeCatps,gradeCat,num
750,14,9,5,NaN,Female,Doing OK,NaN,Grades 9-10,secondary school,1
751,10,5,6,NaN,Male,Doing OK,NaN,Grades 5-6,primary school,2
752,13,7,4,NaN,Male,Struggling,NaN,Grades 7-8,secondary school,3
753,16,11,3,NaN,Gender Diverse,Struggling,NaN,Grades 11-12,secondary school,4
754,12,6,7,NaN,Male,Thriving,NaN,Grades 5-6,primary school,5
755,13,8,8,NaN,Female,Thriving,NaN,Grades 7-8,secondary school,6
756,15,9,6,NaN,Male,Doing OK,NaN,Grades 9-10,secondary school,7
757,15,10,6,NaN,Male,Doing OK,NaN,Grades 9-10,secondary school,8
758,18,12,7,NaN,Female,Thriving,NaN,Grades 11-12,secondary school,9
759,16,11,5,NaN,Male,Doing OK,NaN,Grades 11-12,secondary school,10


In [6]:
missing_year = df[df["year"].isna()]

print("Age range:", missing_year["age"].min(), "-", missing_year["age"].max())
print("Grade range:", missing_year["grade"].min(), "-", missing_year["grade"].max())

print("\nGender:")
print(missing_year["genderN"].value_counts())

print("\nCantril categories:")
print(missing_year["cantrilCat"].value_counts())

Age range: 7 - 18
Grade range: 3 - 12

Gender:
genderN
Female            131
Male              104
Gender Diverse     15
Name: count, dtype: int64

Cantril categories:
cantrilCat
Doing OK      99
Thriving      94
Struggling    57
Name: count, dtype: int64


In [7]:
# Check Cantril category consistency
def expected_cantril_category(score):
    if score <= 4:
        return "Struggling"
    elif score <= 6:
        return "Doing OK"
    else:
        return "Thriving"

df["expected_cantrilCat"] = df["cantril"].apply(expected_cantril_category)

cantril_errors = df[df["cantrilCat"] != df["expected_cantrilCat"]]

print("Cantril category errors:", len(cantril_errors))
display(cantril_errors.head())

Cantril category errors: 0


,age,grade,cantril,ses,genderN,cantrilCat,year,gradeCatps,gradeCat,num,expected_cantrilCat


In [8]:
print("Age range:", df["age"].min(), "-", df["age"].max())
print("Grade range:", df["grade"].min(), "-", df["grade"].max())

print("\nGrade values:")
print(sorted(df["grade"].unique()))

print("\nGrade categories:")
print(df["gradeCatps"].value_counts().sort_index())

print("\nSchool categories:")
print(df["gradeCat"].value_counts())

Age range: 7 - 18
Grade range: 3 - 12

Grade values:
[np.int64(3), np.int64(4), np.int64(5), np.int64(6), np.int64(7), np.int64(8), np.int64(9), np.int64(10), np.int64(11), np.int64(12)]

Grade categories:
gradeCatps
Grades 11-12    102
Grades 3-4      228
Grades 5-6      304
Grades 7-8      202
Grades 9-10     164
Name: count, dtype: int64

School categories:
gradeCat
primary school      532
secondary school    468
Name: count, dtype: int64


In [9]:
def expected_grade_group(grade):
    if grade in [3, 4]:
        return "Grades 3-4"
    elif grade in [5, 6]:
        return "Grades 5-6"
    elif grade in [7, 8]:
        return "Grades 7-8"
    elif grade in [9, 10]:
        return "Grades 9-10"
    elif grade in [11, 12]:
        return "Grades 11-12"

df["expected_gradeCatps"] = df["grade"].apply(expected_grade_group)

grade_group_errors = df[
    df["gradeCatps"] != df["expected_gradeCatps"]
]

print("Grade group errors:", len(grade_group_errors))

Grade group errors: 0


In [10]:
df["expected_school"] = df["grade"].apply(
    lambda x: "primary school" if x <= 6 else "secondary school"
)

school_errors = df[
    df["gradeCat"] != df["expected_school"]
]

print("School category errors:", len(school_errors))

School category errors: 0


In [11]:
print("Gender values:")
print(df["genderN"].value_counts(dropna=False))

print("\nYear values:")
print(df["year"].value_counts(dropna=False))

print("\nDuplicate rows:")
print(df.duplicated().sum())

Gender values:
genderN
Female            493
Male              465
Gender Diverse     42
Name: count, dtype: int64

Year values:
year
2022.0    250
2023.0    250
2024.0    250
NaN       250
Name: count, dtype: int64

Duplicate rows:
0


In [12]:
print("Cantril values:")
print(df["cantril"].value_counts().sort_index())

print("\nCantril range:")
print(df["cantril"].min(), "-", df["cantril"].max())

Cantril values:
cantril
1     12
2      9
3     48
4    115
5    207
6    242
7    186
8    181
Name: count, dtype: int64

Cantril range:
1 - 8


## Validation Summary
- The dataset contains 1,000 rows and 10 columns.
- Age values range from 7 to 18 and grade values range from 3 to 12.
- Cantril scores range from 1 to 8 and all Cantril categories are correct.
- Grade groups and primary/secondary school categories are consistent.
- No duplicate rows were found.
- 250 records have missing year values.
- 327 records have missing SES values.
- All 250 records with missing year values also have missing SES values.
- The missing year and SES values should be confirmed before these records are used in the platform.